packages/playwright-core/src/utils/isomorphic/ariaSnapshot.ts

```diff
@@ -152,6 +153,28 @@ export function parseAriaSnapshot(yaml: YamlLibrary, text: string, options: yaml
         continue;
       }
 
+      // - /strict: deep
+      if (key.value === '/children') {
+        container.containerMode = 'contain';
+        const valueIsString = value instanceof yaml.Scalar && typeof value.value === 'string';
+        if (valueIsString) {
+          if (value.value !== 'contain' && value.value !== 'equal' && value.value !== 'deep-equal') {
+            errors.push({
+              message: 'Strict value should be "contain", "equal" or "deep-equal"',
+              range: convertRange(((entry.value as any).range || map.range)),
+            });
+            continue;
+          }
+
+          if (value.value === 'equal')
```

Comment:
Just unconditional `container.containerMode = value.value;` would do after the check above.

Reply:
it would lose typing.